test/battle/move_effect/dragon_darts.c

```diff
@@ -23,170 +23,148 @@ SINGLE_BATTLE_TEST("Dragon Darts strikes twice")
+    struct BattlePokemon *secondaryTarget = NULL;
+    PARAMETRIZE { chosenTarget = opponentLeft; secondaryTarget = opponentRight; }
+    PARAMETRIZE { chosenTarget = opponentRight; secondaryTarget = opponentLeft; }
+
     GIVEN {
         PLAYER(SPECIES_WOBBUFFET);
         PLAYER(SPECIES_WOBBUFFET);
         OPPONENT(SPECIES_WOBBUFFET);
         OPPONENT(SPECIES_WOBBUFFET);
     } WHEN {
-        TURN { MOVE(playerLeft, MOVE_DRAGON_DARTS, target: opponentLeft); }
+        TURN { MOVE(playerLeft, MOVE_DRAGON_DARTS, target: chosenTarget); }
     } SCENE {
         ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentLeft);
+        HP_BAR(chosenTarget);
         ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentRight);
+        HP_BAR(secondaryTarget);
         MESSAGE("The Pokémon was hit 2 time(s)!");
     }
 }
 
 DOUBLE_BATTLE_TEST("Dragon Darts strikes the ally twice if the target protects")
 {
+    struct BattlePokemon *chosenTarget = NULL;
+    struct BattlePokemon *secondaryTarget = NULL;
+    PARAMETRIZE { chosenTarget = opponentLeft; secondaryTarget = opponentRight; }
+    PARAMETRIZE { chosenTarget = opponentRight; secondaryTarget = opponentLeft; }
     GIVEN {
         PLAYER(SPECIES_WOBBUFFET);
         PLAYER(SPECIES_WOBBUFFET);
         OPPONENT(SPECIES_WOBBUFFET);
         OPPONENT(SPECIES_WOBBUFFET);
     } WHEN {
-        TURN { MOVE(opponentLeft, MOVE_PROTECT); MOVE(playerLeft, MOVE_DRAGON_DARTS, target: opponentLeft); }
-    } SCENE {
-        ANIMATION(ANIM_TYPE_MOVE, MOVE_PROTECT, opponentLeft);
-        ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentRight);
-        ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentRight);
-        MESSAGE("The Pokémon was hit 2 time(s)!");
-    }
-}
-
-DOUBLE_BATTLE_TEST("Dragon Darts strikes the right ally twice if the target is a fairy type")
-{
-    GIVEN {
-        PLAYER(SPECIES_WOBBUFFET);
-        PLAYER(SPECIES_WOBBUFFET);
-        OPPONENT(SPECIES_CLEFAIRY);
-        OPPONENT(SPECIES_WOBBUFFET);
-    } WHEN {
-        TURN { MOVE(playerLeft, MOVE_DRAGON_DARTS, target: opponentLeft); }
+        TURN { MOVE(chosenTarget, MOVE_PROTECT); MOVE(playerLeft, MOVE_DRAGON_DARTS, target: opponentLeft); }
     } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_PROTECT, chosenTarget);
         ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentRight);
+        HP_BAR(secondaryTarget);
         ANIMATION(ANIM_TYPE_MOVE, MOVE_DRAGON_DARTS, playerLeft);
-        HP_BAR(opponentRight);
+        HP_BAR(secondaryTarget);
         MESSAGE("The Pokémon was hit 2 time(s)!");
     }
 }
 
-DOUBLE_BATTLE_TEST("Dragon Darts strikes the left ally twice if the target is a fairy type")
+DOUBLE_BATTLE_TEST("Dragon Darts strikes an opponent twice if the other one is Fairy-type")
 {
+    struct BattlePokemon *chosenTarget = NULL;
```

Comment:
Not something that needs reversion now but could you make big changes like to tests in a separate PR? It is already a lot to review. Not sure what big exactly is but I would classify the Dragon Dart changes as big.

Reply:
eh, it is fine.

Reply:
Sure, I don't have issue of isolating changes to tests themselves to a separate PR if it makes it easier to review. I'll keep it in mind for the next batch